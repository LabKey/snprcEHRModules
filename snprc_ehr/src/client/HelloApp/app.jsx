import React from 'react'
import ReactDOM from 'react-dom'

import App from './HelloApp'

// Need to wait for container element to be available in labkey wrapper before render
window.addEventListener('DOMContentLoaded', (event) => {
    ReactDOM.render(<App title={'Hello, World!'}/>, document.getElementById('app'));
});
