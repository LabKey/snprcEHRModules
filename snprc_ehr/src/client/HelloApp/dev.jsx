import React from 'react'
import ReactDOM from 'react-dom'
import {AppContainer} from 'react-hot-loader'

import App from './HelloApp'

const render = () => {
    ReactDOM.render(
        <AppContainer>
            <App title={'Hello, World!'}/>
        </AppContainer>,
        document.getElementById('app')
    )
};

if (module.hot) {
    module.hot.accept();
}

render();