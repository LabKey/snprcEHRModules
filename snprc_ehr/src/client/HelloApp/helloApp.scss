@import '../styles/styles.scss';

.header_title {
  color: $button-blue;
  text-shadow: 2px 2px $dark-gray;
  font-size: $font-size-xlarge;
  margin-left: 5rem;
}

.header_subtitle {
  color: #333745;
  font-size: 1.6rem;
  font-weight: 500;
  margin-top: 1rem;
  margin-left: 5rem
}