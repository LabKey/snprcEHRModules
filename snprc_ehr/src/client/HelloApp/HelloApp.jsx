import React from 'react';
import "./helloApp.scss";

const HelloApp = (props) => {
     return (
            <div>
                <div className='header_title'>
                    {props.title}
                </div>
                <div className='header_subtitle'>
                    {props.subtitle}
                </div>
            </div>
    )
};

HelloApp.defaultProps = {
    title: 'Hello',
    subtitle: 'Put your life in the hands of a computer.'
};

export default HelloApp;