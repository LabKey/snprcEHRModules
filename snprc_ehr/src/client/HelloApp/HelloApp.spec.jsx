import React from 'react';
import {shallow } from 'enzyme';
import HelloApp from './HelloApp';

describe('<HelloApp/>', () => {
    test('Should render LandingPage correctly.', () => {
        const wrapper = shallow(<HelloApp/>);

        expect(wrapper).toMatchSnapshot();
    });
});