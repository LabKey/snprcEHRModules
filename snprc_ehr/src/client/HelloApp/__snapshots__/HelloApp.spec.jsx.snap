// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloApp/> Should render LandingPage correctly. 1`] = `
<div>
  <div
    className="header_title"
  >
    Hello
  </div>
  <div
    className="header_subtitle"
  >
    Put your life in the hands of a computer.
  </div>
</div>
`;
